x/wasm/simulation/proposals.go

```diff
@@ -78,81 +74,63 @@ func ProposalContents(bk BankKeeper, wasmKeeper WasmKeeper) []simtypes.WeightedP
 				DefaultSimulationCodeIDSelector,
 			),
 		),
-		// simulation.NewWeightedProposalContent(
-		//	WeightSudoContractProposal,
-		//	params.DefaultWeightSudoContractProposal,
-		//	SimulateSudoContractProposal(
-		//		wasmKeeper,
-		//		DefaultSimulateContractSelector,
-		//	),
-		// ),
-		simulation.NewWeightedProposalContent(
+		simulation.NewWeightedProposalMsg(
 			WeightPinCodesProposal,
 			params.DefaultWeightPinCodesProposal,
 			SimulatePinContractProposal(
 				wasmKeeper,
 				DefaultSimulationCodeIDSelector,
 			),
 		),
-		simulation.NewWeightedProposalContent(
+		simulation.NewWeightedProposalMsg(
 			WeightUnpinCodesProposal,
 			params.DefaultWeightUnpinCodesProposal,
 			SimulateUnpinContractProposal(
 				wasmKeeper,
 				DefaultSimulationCodeIDSelector,
 			),
 		),
-		simulation.NewWeightedProposalContent(
+		simulation.NewWeightedProposalMsg(
 			WeightUpdateInstantiateConfigProposal,
 			params.DefaultWeightUpdateInstantiateConfigProposal,
 			SimulateUpdateInstantiateConfigProposal(
 				wasmKeeper,
 				DefaultSimulationCodeIDSelector,
 			),
 		),
-		// simulation.NewWeightedProposalContent(
-		//	WeightStoreAndInstantiateContractProposal,
-		//	params.DefaultWeightStoreAndInstantiateContractProposal,
-		//	SimulateStoreAndInstantiateContractProposal(
-		//		wasmKeeper,
-		//	),
-		// ),
 	}
 }
 
 // simulate store code proposal (unused now)
 // Current problem: out of gas (defaul gaswanted config of gov SimulateMsgSubmitProposal is 10_000_000)
 // but this proposal may need more than it
-//
+func SimulateStoreCodeProposal(wasmKeeper WasmKeeper) simtypes.MsgSimulatorFn {
+	return func(r *rand.Rand, ctx sdk.Context, accs []simtypes.Account) sdk.Msg {
+		// use the default gov module account address as authority
+		var authority sdk.AccAddress = address.Module("gov")
```

Comment:
Nit: nice to see this! To be on the safe side for non gov authority scenarios, you could modify the interface and do:
```go
authority := wasmKeeper.GetAuthority() # returns string
```
Here and other places in this file